[clang][AST] Fix StmtProfile handling of GCCAsmStmt asm strings and clobbers (#201481)
llvm/llvm-project: clang/docs/ReleaseNotes.rst

- Fixed the SourceLocation and SourceRange of reversed rewritten CXXOperatorCallExpr. (#GH192467)
- Fixed a assertion when ``__block`` is used on global variables in C mode. (#GH183974)
- Added missing AST nodes representing the ``decltype`` specifiers in destructor call to AST.

> - Fixed a missing ODR violation diagnostic when two inline definitions of the same function differ only in the inline assembly string or clobber list. (#GH198616)

Reviewer: ```suggestion
- Fixed a missing ODR violation diagnostic introduced by the inline assembly string or clobber list. (#GH198616)
```

The "inline definitions of the same function" part is not accurate.

> - Fixed ``GCCAsmStmt`` profiling so that inline asm statements differing only in their asm string or clobber list no longer compare equal in ``Stmt::Profile`` and the ODR hash. (#GH198616)

Reviewer: Release note is for end users. They won't know what is ``Stmt::Profile`` or ODR Hash. It is better to say "the ODR violation caused by ... can be diagnosed correctly".

Author: I made the release note more user-friendly.
Thanks for your feedback! 👍
